src/app/core/parameter.ts

```diff
@@ -64,52 +68,21 @@ export default class Parameter {
      * @memberof Parameter
      * @returns {void}
      */
-    updateValue(value) {
-        Parameter.checkValue(this.__type, value);
-        this.__value = value;
+    updateValue(value: any) {
+        Parameter.checkValue(this.type, value);
+        this.value = value;
     }
     
     resetValue() {}
 
-    //Takes a typestring to recognize that param type, and
-    // an isValid function which returns true if a value is OK for
-    // that type.
-    /**
-     * Registers a parameter
-     * @param {String} typeString Type of string ?
-     * @param {Boolean} isValid Boolean to asure the parameter is valid
-     * @param {String} description Description of the parameter
-     * @memberof Parameter
-     * @returns {void}
-     */
-    static registerParamType(typeString, isValid, description) {
-        Registry.registeredParams[typeString] = {
-            isValid: isValid,
-            description: description
-        };
-    }
-    /**
-     * Creates a new type of parameter with a specified value
-     * @param {String} type Type of the new parameter
-     * @param {*} value Value of the new parameter
-     * @returns {Parameter} Returns a parameter object
-     * @memberof Parameter
-     */
-    static makeParam(type, value) {
-        if (Registry.registeredParams.hasOwnProperty(type)) {
-            return new Parameter(type, value);
-        } else {
-            throw new Error("Type " + type + " has not been registered.");
-        }
-    }
     /**
      * Creates a parameter from a JSON format
      * @param {JSON} json JSON format file with the parameters loaded
      * @returns {Parameter} Returns a new parameter
      * @memberof Parameter
      */
-    static fromJSON(json) {
-        return Parameter.makeParam(json.type, json.value);
+    static fromJSON(json: any) {
+        return new Parameter(json.type, json.value);
```

Comment:
We need to modify this into a new static method that will check what type of json value is and then create the corresponding subclass